js/lib/ui.js

```diff
@@ -418,11 +418,23 @@ elgg.ui.initAccessInputs = function () {
 		}
 		var $select = $(this),
 			acl = $select.data('group-acl'),
-			$note = $('.elgg-input-access-membersonly', this.parentNode);
+			$note = $('.elgg-input-access-membersonly', this.parentNode),
+			$commentCount = $select.data('comment-count'),
```

Comment:
Nevermind, it does work - I tested on an entity without comments.
It works both ways, likely jquery normalizing it.  I'm going to leave it as-is for now because it's likely there are lots of instances of this.  If we want to be strict about it we can do a separate ticket to audit the js usage of data.